library/std/src/panicking.rs

```diff
@@ -74,7 +74,7 @@ enum Hook {
     Custom(*mut (dyn Fn(&PanicInfo<'_>) + 'static + Sync + Send)),
 }
 
-static HOOK_LOCK: RWLock = RWLock::new();
+static HOOK_LOCK: StaticRWLock = StaticRWLock::new();
```

Comment:
`panicking` currently uses `sys_common::RWLock`, but could also use something like `SyncOnceCell<sync::RwLock<Hook>>` which would require less unsafe. Is there a reason to prefer `StaticRWLock` outside of `sys`? If not then `StaticRWLock` could be kept for internal use in `sys` only and moved to `sys::common`. Related comment: https://github.com/rust-lang/rust/pull/84519#issuecomment-827634965